fuel-txpool/src/service.rs

```diff
@@ -1,59 +1,191 @@
-use crate::{interface::Interface, Config};
+use crate::{Config, TxPool};
+use anyhow::anyhow;
 use fuel_core_interfaces::block_importer::ImportBlockBroadcast;
-use fuel_core_interfaces::txpool::{Sender, TxPoolDb, TxPoolMpsc, TxStatusBroadcast};
+use fuel_core_interfaces::txpool::{self, TxPoolDb, TxPoolMpsc, TxStatusBroadcast};
 use std::sync::Arc;
-use tokio::sync::{broadcast, mpsc, Mutex};
+use tokio::sync::{broadcast, mpsc, Mutex, RwLock};
 use tokio::task::JoinHandle;
-use tracing::warn;
 
-pub struct Service {
-    interface: Arc<Interface>,
-    sender: Sender,
+pub struct ServiceBuilder {
+    sender: txpool::Sender,
+    receiver: mpsc::Receiver<TxPoolMpsc>,
+    config: Config,
     broadcast: broadcast::Sender<TxStatusBroadcast>,
-    join: Mutex<Option<JoinHandle<mpsc::Receiver<TxPoolMpsc>>>>,
-    receiver: Arc<Mutex<Option<mpsc::Receiver<TxPoolMpsc>>>>,
+    db: Option<Box<dyn TxPoolDb>>,
+    import_block_events: Option<broadcast::Receiver<ImportBlockBroadcast>>,
 }
 
-impl Service {
-    pub fn new(db: Box<dyn TxPoolDb>, config: Config) -> anyhow::Result<Self> {
+impl Default for ServiceBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ServiceBuilder {
+    pub fn new() -> Self {
         let (sender, receiver) = mpsc::channel(100);
         let (broadcast, _receiver) = broadcast::channel(100);
+        Self {
+            sender: txpool::Sender::new(sender),
+            receiver,
+            db: None,
```

Comment:
It is a service builder, every setter is made so that it can be cross-connected with other modules, it is good to have same logic for all fields and setting db from `new` is going to break that.

Default to be `InMemory` db seems interesting idea, but i feel it is better for a caller to be explicit on what they want to be set.